### ISSUE TITLE
[Discussion] Automaton making is missing 

### ISSUE THREAD

**Issue by sribharghava** [vi3k6i5/flashtext#34]
Hi, 

I'm a newbie to github and I don't see any discussion forum here so I'm opening the discussion in an Issue. Excuse me for my ignorance. 

The paper that you pointed out says that the flashtext's algorithm is different from aho-corasick.
I've a few questions on this statement. Here's what I understand!!
You're adding start and end tags for keywords to avoid match at character level and you're considering only the longest match. 
But aren't you actually missing out on the core part of the ahocorasick algorithm which is automaton building. 

Based on my understanding referencing ahocorasick in the paper is puzzling as you're just building a Trie and using it for retreival and not actually building an automaton out of that trie.

Please correct me if my understanding is wrong

**Comment by vi3k6i5** [comment]
You are very right. That is exactly what I have done.

I have built a trie and made sure I always go for longest match, and
partial matches are not considered.

I referenced Aho corasic because that's where I got the idea from. And
FlashText is inspired by it. So for me, it had to be there. But I am not
creating an automaton, no. I didn't had to, I got the performance I wanted,
without automaton.

Plus a few advantages, like replace word with clean term. As clean term is
also kept in dictionary.

Hope this answers your question. Plus it's fine to create issues for
questions. As I don't have a discussion forum yet. :D.
